src/Search.js

```diff
@@ -147,17 +156,31 @@ class Search extends React.Component {
     },
   });
 
-  onChangeIndex = (e) => {
-    const qindex = e.target.value;
-    const logger = this.props.stripes.logger;
-    logger.log('action', `changed query-index to '${qindex}'`);
-    this.props.mutator.query.update({ qindex });
-  }
+  onFilterChangeHandler = ({ name, values }) => {
+    const filters = parseFiltersString(_.get(this.props, 'resources.query.filters', ''));
+
+    const updatedActiveFilters = values.length !== 0
+      ? {
+        ...filters,
+        [name]: values,
+      }
+      : _.omit(filters, [name]);
+
+    const newFiltersString = Object.keys(updatedActiveFilters).map((filterName) => {
+      return updatedActiveFilters[filterName].map((filterValue) => {
```

Comment:
Could we use `arr.reduce` here?

Reply:
In a perfect world that would be true, but here it's necessary to add a condition for tracking the comma. The way I see it's better to use `for in` construction.